kotsadm/pkg/handlers/login.go

```diff
@@ -63,3 +97,239 @@ func Login(w http.ResponseWriter, r *http.Request) {
+			return
+		}
+	default:
+		logger.Error(errors.Errorf("method not implemented: %s", r.Method))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	rawIDToken, ok := token.Extra("id_token").(string)
+	if !ok {
+		logger.Error(errors.New("no id_token in token response"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	verifier := provider.Verifier(&oidc.Config{ClientID: oauth2Config.ClientID})
+	idToken, err := verifier.Verify(r.Context(), rawIDToken)
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to verify ID token"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var claims struct {
+		Email    string   `json:"email"`
+		Name     string   `json:"name"`
+		Verified bool     `json:"email_verified"`
+		Groups   []string `json:"groups"`
+	}
+	if err := idToken.Claims(&claims); err != nil {
+		logger.Error(errors.Wrap(err, "error decoding ID token claims"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	user := &usertypes.User{
+		ID: claims.Email,
+	}
+
+	identityConfig, err := identity.GetConfig(r.Context(), os.Getenv("POD_NAMESPACE"))
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to get identity config"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	roles := []sessiontypes.SessionRole{}
+	if identityConfig.EnableAdvancedRBAC {
+		roles = session.GetSessionRolesFromRBAC(claims.Groups, identityConfig.RBAC.Groups, identityConfig.RBAC.Roles, identityConfig.RBAC.Policies)
+	} else {
+		groups := rbac.DefaultGroups
+		if len(identityConfig.RestrictedGroups) > 0 {
+			groups = identity.RestrictedGroupsToRBACGroups(identityConfig.RestrictedGroups)
+		}
+		roles = session.GetSessionRolesFromRBAC(claims.Groups, groups, rbac.DefaultRoles, rbac.DefaultPolicies)
+	}
+
+	createdSession, err := store.GetStore().CreateSession(user, &idToken.Expiry, roles)
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to create session"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	signedJWT, err := session.SignJWT(createdSession)
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to sign jwt"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	responseToken := fmt.Sprintf("Bearer %s", signedJWT)
+
+	ingressConfig, err := ingress.GetConfig(r.Context(), os.Getenv("POD_NAMESPACE"))
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to get ingress config"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// TODO: support tls?
+	expire := time.Now().Add(30 * time.Minute)
+	cookie := http.Cookie{
+		Name:    "token",
+		Value:   responseToken,
+		Expires: expire,
+		Path:    "/",
+	}
+	http.SetCookie(w, &cookie)
+
+	http.Redirect(w, r, ingress.GetAddress(*ingressConfig), http.StatusSeeOther)
+}
+
+type GetLoginInfoResponse struct {
+	Method            LoginMethod `json:"method"`
+	IdentityConnector string      `json:"identityConnector,omitempty"` // TODO: support multiple connectors
```

Comment:
Can this contain sensitive info?

Reply:
this handler has been removed in this pr https://github.com/replicatedhq/kots/pull/1275

Reply:
and no it won't contain any sensitive info because we only return the connector name. that's a string field